api/models.py

```diff
@@ -297,6 +297,10 @@ class ExpiryReasons(models.IntegerChoices):
         NESINV = 4, "Neither escrow locked or invoice submitted"
 
     # order info
+    reference = models.CharField(max_length=36,
```

Comment:
Might be cleaner if we use a `UUIDField` instead of a `CharField`. It will be filled with a UUID as soon as the object is created  (more info on UUIDFields https://www.geeksforgeeks.org/uuidfield-django-models/ )
```python
import uuid
  
reference = models.UUIDField(
         default = uuid.uuid4,
         editable = False)
```

Would save a few lines also in views.py